src/content/1.7/modules/sample_modules/grid-and-identifiable-object-form-hooks-usage.md

```diff
@@ -0,0 +1,227 @@
+use PrestaShopBundle\Form\Admin\Type\YesAndNoChoiceType;
+
+/**
+     * Hook allows to modify Customers grid definition.
+     * This hook is a right place to add/remove columns or actions (bulk, grid).
+     *
+     * @param array $params
+     */
+    public function hookActionCustomerGridDefinitionModifier(array $params)
+    {
+        /** @var GridDefinitionInterface $definition */
+        $definition = $params['definition'];
+
+        $translator = $this->getTranslator();
+
+        $definition
+            ->getColumns()
+            ->addAfter(
+                'optin',
+                (new ToggleColumn('is_allowed_for_review'))
+                    ->setName($translator->trans('Allowed for review', [], 'Modules.Ps_DemoCQRSHooksUsage'))
+                    ->setOptions([
+                        'field' => 'is_allowed_for_review',
+                        'primary_field' => 'id_customer',
+                        'route' => 'ps_democqrshooksusage_toggle_is_allowed_for_review',
+                        'route_param_name' => 'customerId',
+                    ])
+            )
+        ;
+
+        $definition->getFilters()->add(
+            (new Filter('is_allowed_for_review', YesAndNoChoiceType::class))
+            ->setAssociatedColumn('is_allowed_for_review')
+        );
+    }
+```
+
+In this sample a new toggable column which determines if the customer is eligable to review products is added just after another column which has id `optin`. The sample code also demonstrates how add new filter.
+
+#### Creating route for toggle column
+
+As in this sample module we are creating `ToggleColumn` we need to configure the route in which the toggling action will be performed.
+
+{{% notice note %}}
+
+ If you only want to display data then this step can be skipped. E.g you are creating `DataColumn`
+
+{{% /notice %}}
+
+- Create controller `DemoCQRSHooksUsage\Controller\Admin\CustomerReviewController`:
+
+```php
+namespace DemoCQRSHooksUsage\Controller\Admin;
+
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+
+class CustomerReviewController extends FrameworkBundleAdminController
+{
+}
+```
+
+- Create controller `action`:
+
+{{% notice note %}}
+**This example has been simplified for practical reasons.** 
+
+You can find full implementation [here](https://github.com/friends-of-prestashop/demo-cqrs-hooks-usage-module) which uses CQRS pattern to toggle the reviewer state. [More about it here]({{< relref "cqrs.md" >}}).
+{{% /notice %}}
+
+```php
+namespace DemoCQRSHooksUsage\Controller\Admin;
+
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+
+class CustomerReviewController extends FrameworkBundleAdminController
+{
+    public function toggleIsAllowedForReviewAction($customerId)
+    {
+        //  updating reviewer state can be handled here
+        return $this->redirectToRoute('admin_customers_index');
+    }
+}
+```
+
+- Create a route in `ps_democqrshooksusage/config/routes.yml` file:
```

Comment:
```suggestion
- As this is a Symfony controller, we must configure the related routing, which means create a route in `ps_democqrshooksusage/config/routes.yml` file:
```

Reply:
Maybe with a link to https://symfony.com/doc/current/routing.html

Reply:
It seems for me that putting links to symfony is too much. I think a person which is not familiar with it will google about it because in the text above he has key phrases such as `Symfony` and `routing`.

Reply:
but I saw in docs we already doing that so never mind - I'll put a link here as well :D